### ISSUE TITLE
class-level instance variables

### ISSUE THREAD

**Issue by skippy** [localshred/protobuf#246]
hey folks,

I'm looking at using the base protobuf class and wrapping business logic around it in a separate class, using class inheritance.  However, this doesn't work in this library:
```ruby
require 'protobuf'
require 'protobuf/message'
module Corp
  module Protobuf
    class Error < ::Protobuf::Message
      required :string, :foo, 1
    end
  end
end
module Corp
  class ErrorHandler < Corp::Protobuf::Error
  end
end
# this works
Corp::Protobuf::Error.new(foo: :bar).encode #=> "\n\x03bar"
#this does not
Corp::ErrorHandler.new(foo: :bar).encode #=> ""
```

The root issue is within `lib/protobuf/message/fields.rb` and the fact that there is a somewhat hidden class instance variable `@field_store` and a bunch of class instance variables uses for memoization.  The module doesn't define it as a class instance variable, obviously, but it extends the message class, which makes it a class instance variable.

Is this intended behavior?  Do you think it is worth fixing?  If so, and since active_support is already included, I think `mattr_accessor` will do the trick (I can submit a patch and find out :)  And of course I can change my class to use the protobuf as an internal object and forgo inheritance all together.

thoughts?  And thank you for the library!

**Comment by localshred** [comment]
@skippy definitely not intended. I extracted a bunch of class-level methods from `Message` about a year ago and obviously didn't have a test for what you're doing. Would love to see a PR fixing it. :+1:

**Comment by localshred** [comment]
Fixed under [v3.4.4](https://github.com/localshred/protobuf/releases/tag/v3.4.4).

**Comment by skippy** [comment]
thanks!
